### ISSUE TITLE
istio v1.5.1 fails with failed to reconcile resource: creating resource failed

### ISSUE THREAD

**Issue by Starefossen** [banzaicloud/istio-operator#416]
**Describe the bug**

Fresh install of Istio v1.5.1 with latest version of operator (v0.5.5) on GKE gives the following error:

```yaml
  status:
    ErrorMessage: |-
      failed to reconcile resource: creating resource failed: clusterroles.rbac.authorization.k8s.io "istio-cni-repair-role" is forbidden: user "system:serviceaccount:istio-system:istio-operator" (groups=["system:serviceaccounts" "system:serviceaccounts:istio-system" "system:authenticated"]) is attempting to grant RBAC permissions not currently held:
      {APIGroups:[""], Resources:["events"], Verbs:["get" "list" "watch" "delete" "update"]}
      {APIGroups:[""], Resources:["pods"], Verbs:["delete" "patch" "update"]}
    Status: ReconcileFailed
```

**Steps to reproduce the issue:**

Apply the following istio config:

```yaml
  apiVersion: istio.banzaicloud.io/v1beta1
  kind: Istio
  metadata:
    creationTimestamp: "2020-04-16T17:20:18Z"
    finalizers:
    - istio-operator.finializer.banzaicloud.io
    generation: 2
    labels:
      app.kubernetes.io/instance: istio
      app.kubernetes.io/managed-by: Helm
      app.kubernetes.io/name: istio-gke
      controller-tools.k8s.io: "1.0"
      helm.sh/chart: istio-gke-1.4.0
    name: istio-istio-gke
    namespace: istio-system
    resourceVersion: "852556"
    selfLink: /apis/istio.banzaicloud.io/v1beta1/namespaces/istio-system/istios/istio-istio-gke
    uid: 488779d6-3dff-4d45-9e1e-c6747bf16911
  spec:
    autoInjectionNamespaces:
    - default
    autoMtls: true
    citadel:
      enableNamespacesByDefault: true
      enabled: false
      image: docker.io/istio/citadel:1.5.1
    clusterName: Kubernetes
    defaultPodDisruptionBudget:
      enabled: false
    defaultResources:
      requests:
        cpu: 10m
    galley:
      configValidation: true
      enableAnalysis: false
      enableServiceDiscovery: false
      enabled: false
      image: docker.io/istio/galley:1.5.1
      replicaCount: 1
    gateways:
      egress:
        enabled: false
        maxReplicas: 5
        minReplicas: 1
        ports:
        - name: http2
          port: 80
          protocol: TCP
          targetPort: 80
        - name: https
          port: 443
          protocol: TCP
          targetPort: 443
        - name: tls
          port: 15443
          protocol: TCP
          targetPort: 15443
        replicaCount: 1
        sds:
          enabled: false
          image: docker.io/istio/node-agent-k8s:1.5.1
        serviceType: ClusterIP
      enabled: true
      ingress:
        enabled: true
        maxReplicas: 5
        minReplicas: 1
        ports:
        - name: status-port
          port: 15020
          protocol: TCP
          targetPort: 15020
        - name: http2
          port: 80
          protocol: TCP
          targetPort: 80
        - name: https
          port: 443
          protocol: TCP
          targetPort: 443
        - name: tls
          port: 15443
          protocol: TCP
          targetPort: 15443
        - name: tcp
          port: 31400
          protocol: TCP
          targetPort: 31400
        replicaCount: 1
        sds:
          enabled: false
          image: docker.io/istio/node-agent-k8s:1.5.1
        serviceType: LoadBalancer
      k8singress:
        enableHttps: false
        enabled: false
    imagePullPolicy: IfNotPresent
    includeIPRanges: '*'
    istioCoreDNS:
      enabled: false
      image: coredns/coredns:1.6.2
      pluginImage: docker.io/istio/coredns-plugin:0.2-istio-1.1
      replicaCount: 1
    istiod:
      enabled: true
    jwtPolicy: first-party-jwt
    logging:
      level: default:info
    meshExpansion: false
    meshPolicy:
      mtlsMode: PERMISSIVE
    mixer:
      enabled: false
      image: docker.io/istio/mixer:1.5.1
      maxReplicas: 5
      minReplicas: 1
      replicaCount: 1
      reportBatchMaxEntries: 100
      reportBatchMaxTime: 1s
      sessionAffinityEnabled: false
      stdioAdapterEnabled: false
    mixerlessTelemetry:
      enabled: true
    mountMtlsCerts: false
    multiMesh: false
    networkName: local-network
    nodeAgent:
      enabled: false
      image: docker.io/istio/node-agent-k8s:1.5.1
    outboundTrafficPolicy:
      mode: ALLOW_ANY
    pilot:
      certProvider: istiod
      enableProtocolSniffingInbound: false
      enableProtocolSniffingOutbound: true
      enabled: true
      image: docker.io/istio/pilot:1.5.1
      maxReplicas: 5
      minReplicas: 1
      replicaCount: 1
      sidecar: true
      traceSampling: 1
    policy:
      checksEnabled: false
      enabled: false
      image: docker.io/istio/mixer:1.5.1
      maxReplicas: 5
      minReplicas: 1
      replicaCount: 1
    proxy:
      accessLogEncoding: TEXT
      accessLogFile: /dev/stdout
      accessLogFormat: ""
      clusterDomain: cluster.local
      componentLogLevel: misc:error
      coreDumpImage: busybox
      dnsRefreshRate: 300s
      enableCoreDump: false
      envoyAccessLogService:
        enabled: false
        tcpKeepalive:
          interval: 10s
          probes: 3
          time: 10s
        tlsSettings:
          mode: DISABLE
      envoyMetricsService:
        enabled: false
        tcpKeepalive:
          interval: 10s
          probes: 3
          time: 10s
        tlsSettings:
          mode: DISABLE
      envoyStatsD:
        enabled: false
      image: docker.io/istio/proxyv2:1.5.1
      lifecycle: {}
      logLevel: warning
      protocolDetectionTimeout: 100ms
      resources:
        limits:
          cpu: "2"
          memory: 1Gi
        requests:
          cpu: 100m
          memory: 128Mi
      useMetadataExchangeFilter: false
    proxyInit:
      image: docker.io/istio/proxyv2:1.5.1
    proxyWasm:
      enabled: false
    sds:
      enabled: false
      tokenAudience: istio-ca
      udsPath: unix:/var/run/sds/uds_path
    sidecarInjector:
      autoInjectionPolicyEnabled: true
      enabled: true
      image: docker.io/istio/sidecar_injector:1.5.1
      init:
        resources:
          limits:
            cpu: 100m
            memory: 50Mi
          requests:
            cpu: 10m
            memory: 10Mi
      initCNIConfiguration:
        binDir: /home/kubernetes/bin
        chained: true
        confDir: /etc/cni/net.d
        enabled: true
        excludeNamespaces:
        - istio-system
        image: docker.io/istio/install-cni:1.5.1
        logLevel: info
        repair:
          brokenPodLabelKey: cni.istio.io/uninitialized
          brokenPodLabelValue: "true"
          deletePods: true
          enabled: true
          hub: ""
          initContainerName: istio-validation
          labelPods: true
          tag: ""
      replicaCount: 1
    telemetry:
      enabled: false
      image: docker.io/istio/mixer:1.5.1
      maxReplicas: 5
      minReplicas: 1
      replicaCount: 1
      reportBatchMaxEntries: 100
      reportBatchMaxTime: 1s
      sessionAffinityEnabled: false
    tracing:
      datadog: {}
      enabled: true
      lightstep: {}
      stackdriver: {}
      tracer: zipkin
      zipkin:
        address: zipkin.istio-system:9411
    trustDomain: cluster.local
    useMCP: false
    version: 1.5.1
```

**Expected behavior**

Expected istio to start up without errors.

**Screenshots**

N/A

**Additional context**

```
Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.11", GitCommit:"d94a81c724ea8e1ccc9002d89b7fe81d58f89ede", GitTreeState:"clean", BuildDate:"2020-03-12T21:08:59Z", GoVersion:"go1.12.17", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"15+", GitVersion:"v1.15.9-gke.26", GitCommit:"525ce678faa2b28483fa9569757a61f92b7b0988", GitTreeState:"clean", BuildDate:"2020-03-06T18:47:39Z", GoVersion:"go1.12.12b4", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by Laci21** [comment]
@Starefossen We have just released Docker image version `0.5.6` and Helm chart version `chart/istio-operator/0.0.46` with the fix. Enjoy!

**Comment by Starefossen** [comment]
Thanks a lot ❤️
